app/src/main/java/com/github/fribourgsdp/radio/data/Playlist.kt

```diff
@@ -112,6 +120,31 @@ data class Playlist (override var name: String, var genre: Genre) : Nameable {
         return Tasks.whenAll(songTask,playlistId)
     }
 
+    /**
+     * Loads the lyrics of all the songs of the playlist that don't have lyrics
+     * @param [lyricsGetter] the lyricsGetter that provides the lyrics, by default Musixmatch
+     */
+    fun loadLyrics(lyricsGetter: LyricsGetter = MusixmatchLyricsGetter) : Task<Void>{
+
+        val tasks = mutableListOf<Task<Void>>()
+        for (song in songs){
+            if(song.lyrics == "") {
+                tasks.add(
+                    Tasks.forResult(lyricsGetter.getLyrics(song.name, song.artist).thenAccept {
+                        song.lyrics = it
+                    }.join())
```

Comment:
Does this join() make this method blocking ?

Reply:
No, as the join is withing another asynchronous object, the Task.